lib/realtime/incoming_vehicles.ex

```diff
@@ -0,0 +1,103 @@
+defmodule Realtime.IncomingVehicles do
```

Comment:
I had a hard time getting a handle on the new flow of things and I'm trying to figure out how to make that concrete.

One thing that comes to mind is that a bunch of naming uses "incoming" in a way that I'm not sure clarifies as well as they could.

For one example `IncomingVehicles` really seems like it's more about taking all vehicles in and partitioning them into "on route" and "incoming" as equal partners on the way out. I'm not sure exactly what a better name here would be, `PartitionedVehicles` comes to mind but I'm not sure if that's as descriptive as it could be.

This is similarly reflected in the type `vehicles_with_incoming`. For me this doesn't really reflect the structure and intent of the data inside.

Reply:
Yeah, I agree. The way it currently is, it's like once they're partitioned, the non-incoming vehicles become the responsibility of this module, which is named as if it's just responsible for incoming vehicles, and that doesn't feel quite right.

`PartitionedVehicles` gets across the structure of the data, and feels like an appropriate name to be responsible for that structure, but doesn't really say what they're partitioned by.

I'll try to find a better name for that `%{on_route, incoming}` type.